### PR TITLE
Transfered import problem tests logic from old judge and fixed bugs

### PR DESCRIPTION
**Pull request by shsimeonova**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/256

### CODE REVIEW

**Review by mishogv — APPROVED** [review]

**Review by mishogv — DISMISSED** [review]

File: Services/Administration/OJS.Services.Administration.Business/Implementations/ZippedTestsParserService.cs [hunk 1]
```diff
@@ -39,10 +42,122 @@ public async Task<TestsParseResult> Parse(Stream stream)
...
+                IsTrialTest = true,
+                OrderBy = zeroTestsOrder,
+                Problem = problem,
+                InputDataAsString = GetInputData(zeroInput, tryOptimizeMysqlQuery),
+                OutputDataAsString = tests.ZeroOutputs[index],
+            });
+
+            zeroTestsOrder++;
+            addedTestsCount++;
+        });
```
Comment (Minkov) [inline Services/Administration/OJS.Services.Administration.Business/Implementations/ZippedTestsParserService.cs:75]: This code is repeated 3 times. Can be moved into a method.

File: Services/Administration/OJS.Services.Administration.Business/Implementations/ZippedTestsParserService.cs [hunk 2]
```diff
@@ -39,10 +42,122 @@ public async Task<TestsParseResult> Parse(Stream stream)
...
+            problem.Tests.Add(new Test
+            {
+                IsTrialTest = true,
+                OrderBy = zeroTestsOrder,
+                Problem = problem,
+                InputDataAsString = GetInputData(zeroInput, tryOptimizeMysqlQuery),
+                OutputDataAsString = tests.ZeroOutputs[index],
+            });
+
+            zeroTestsOrder++;
```
Comment (Minkov) [inline Services/Administration/OJS.Services.Administration.Business/Implementations/ZippedTestsParserService.cs:73]: I see no reason for this counter. Index should do the same.

File: Services/Administration/OJS.Services.Administration.Business/Implementations/ZippedTestsParserService.cs [hunk 3]
```diff
@@ -39,10 +42,122 @@ public async Task<TestsParseResult> Parse(Stream stream)
...
+        tests.ZeroInputs.ForEach((index, zeroInput) =>
+        {
+            problem.Tests.Add(new Test
+            {
+                IsTrialTest = true,
+                OrderBy = zeroTestsOrder,
+                Problem = problem,
+                InputDataAsString = GetInputData(zeroInput, tryOptimizeMysqlQuery),
+                OutputDataAsString = tests.ZeroOutputs[index],
+            });
```
Comment (Minkov) [inline Services/Administration/OJS.Services.Administration.Business/Implementations/ZippedTestsParserService.cs:71]: `problem.Tests.AddRange(tests.ZeroInputs.Select(...))`

**Review by Minkov — CHANGES_REQUESTED** [review]

File: Services/Administration/OJS.Services.Administration.Business/Implementations/ZippedTestsParserService.cs (hunk 3 above)
Comment (shsimeonova) [reply Services/Administration/OJS.Services.Administration.Business/Implementations/ZippedTestsParserService.cs]: This does not apply here as i need the `index ` variable in my logic

**Review by Minkov — APPROVED** [review]

**Review by mishogv — APPROVED** [review]

**Pull request merged**